segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/ContainerRecoveryUtils.java

```diff
@@ -242,4 +245,39 @@ private static void validateContainerIds(Map<Integer, DebugStreamSegmentContaine
                 storage.openWrite(segmentName).thenCompose(segmentHandle -> storage.delete(segmentHandle, TIMEOUT)),
                 ex -> ex instanceof StreamSegmentNotExistsException, null);
     }
+
+    /**
+     * Creates a target segment with the given name and copies the contents of the source segment to the target segment.
+     * @param storage                   A storage instance to create the segment.
+     * @param sourceSegment             The name of the source segment to copy the contents from.
+     * @param targetSegment             The name of the segment to write the contents to.
+     * @param executor                  A thread pool for execution.
+     * @return                          A CompletableFuture that, when completed normally, will indicate the operation
+     * completed. If the operation failed, the future will be failed with the causing exception.
+     */
+    protected static CompletableFuture<Void> copySegment(Storage storage, String sourceSegment, String targetSegment, ExecutorService executor) {
+        AtomicInteger offset = new AtomicInteger();
+        AtomicInteger bytesToRead = new AtomicInteger();
+        return storage.create(targetSegment, TIMEOUT).thenCompose(targetHandle -> {
```

Comment:
What happens when the segment copy is interrupted in the middle (by whatever reason, including a crash)?

From what I can tell, the target segment will be left in an inconsistent state. It will either be empty or have some partial data in it. 

Should we change this method to either:
1. Delete and recreate the segment (i.e., overwrite)
2. Throw (like now)
3. Resume from where it left off.

If think #3 is off the table given that it may already contain some other, unrelated data. I would choose between 1 and 2. Does the upstream code handle an exception gracefully? Can it recover from it?

Reply:
This method will be used in an upcoming PR(#5180). Since, delete & re-copy can be handled there. 
I will be working on renaming the segments, then this method will not be used.